src/test/kotlin/fi/metatavu/muisti/api/test/functional/PageLayoutTestsIT.kt

```diff
@@ -34,15 +34,68 @@ class PageLayoutTestsIT: AbstractFunctionalTest() {
+
+            val defaultPageLayout = PageLayout(
+                    name = "created name",
+                    data = createdData,
+                    thumbnailUrl = "http://example.com/thumbnail.png",
+                    screenOrientation = ScreenOrientation.portrait,
+                    modelId = createdModelId
+            )
+
+            val createdPageLayout = it.admin().pageLayouts().create(defaultPageLayout)
+            val secondCreatedPageLayout = it.admin().pageLayouts().create(defaultPageLayout)
+            val thirdCreatedPageLayout = it.admin().pageLayouts().create(defaultPageLayout)
+
+            val fourthCreatedPageLayout = it.admin().pageLayouts().create(PageLayout(
+                    name = "created name",
+                    data = createdData,
+                    thumbnailUrl = "http://example.com/thumbnail.png",
+                    screenOrientation = ScreenOrientation.landscape,
+                    modelId = anotherCreatedModelId
+            ))
+
+            val allPageLayouts = it.admin().pageLayouts().listPageLayouts()
+            assertEquals(4, allPageLayouts.size)
+
+            val pageLayoutsByDeviceModelIdAndOrientation = it.admin().pageLayouts().listPageLayoutsByDeviceModelIdAndOrientation(createdModelId, ScreenOrientation.portrait)
+            assertEquals(3, pageLayoutsByDeviceModelIdAndOrientation.size)
+
+            val pageLayoutsByDeviceModelIdAndOrientation2 = it.admin().pageLayouts().listPageLayoutsByDeviceModelIdAndOrientation(anotherCreatedModelId, ScreenOrientation.landscape)
+            assertEquals(1, pageLayoutsByDeviceModelIdAndOrientation2.size)
+
+            val pageLayoutsByDeviceModelIdAndIncorrectOrientation = it.admin().pageLayouts().listPageLayoutsByDeviceModelIdAndOrientation(createdModelId, ScreenOrientation.landscape)
+            assertEquals(0, pageLayoutsByDeviceModelIdAndIncorrectOrientation.size)
+
+            val pageLayoutsByIncorrectDeviceModelId = it.admin().pageLayouts().listPageLayoutsByDeviceModelIdAndOrientation(UUID.randomUUID(), ScreenOrientation.portrait)
+            assertEquals(0, pageLayoutsByIncorrectDeviceModelId.size)
+
+            val pageLayoutsById = it.admin().pageLayouts().listPageLayoutsByDeviceModelId(createdModelId)
+            assertEquals(3, pageLayoutsById.size)
+
+            val pageLayoutsByIncorrectDeviceId = it.admin().pageLayouts().listPageLayoutsByDeviceModelId(UUID.randomUUID())
+            assertEquals(0, pageLayoutsByIncorrectDeviceId.size)
+
+            val pageLayoutsByOrientationPortrait = it.admin().pageLayouts().listPageLayoutsByOrientation(ScreenOrientation.portrait)
+            assertEquals(3, pageLayoutsByOrientationPortrait.size)
+
+            val pageLayoutsByOrientationLandscape = it.admin().pageLayouts().listPageLayoutsByOrientation(ScreenOrientation.landscape)
+            assertEquals(1, pageLayoutsByOrientationLandscape.size)
+
+            it.admin().pageLayouts().delete(createdPageLayout.id!!)
```

Comment:
TestBuilder uses FIFO queue to clean it self up, so if you arrange operations properly, these deletes are unnecessary